dispatcher/msgdispatcher.go

```diff
@@ -144,15 +144,22 @@ var (
 		defer genericPanicRecoveryFunc()
 		jobs := msgDispatcher.djRepo.GetJobsReadyForInflightSince(msgDispatcher.rationalDelay)
 		for _, job := range jobs {
-			if job.Listener.Type != data.PushConsumer {
-				continue
-			}
-			err := inLockRun(msgDispatcher.lockRepo, job, func() error {
-				queueJob(msgDispatcher, job)
-				return nil
-			})
-			if err != nil {
-				log.Error().Err(err).Msg("error - could not retry job" + job.ID.String())
+
+			if job.Listener.Type == data.PullConsumer {
+				if job.RetryAttemptCount >= uint(msgDispatcher.brokerConfig.GetMaxRetry()) {
+					err := inLockRun(msgDispatcher.lockRepo, job, func() error {
+						return msgDispatcher.djRepo.MarkJobDead(job)
```

Comment:
Here, `GetJobsReadyForInflightSince()` fetches the jobs in `JobQueued` status. Then `MarkJobDead()` tries to shift it from `JobInflight` to `JobDead`, but that doesn't work. So the job stays in `JobQueued`. 😢